exercises/practice/anagram/test/anagram_test.dart

```diff
@@ -1,31 +1,21 @@
-import 'package:anagram/anagram.dart';
 import 'package:test/test.dart';
+import 'package:anagram/anagram.dart';
 
 final anagram = Anagram();
 
 void main() {
-  group("simple tests", simpleTests);
-
-  group("slightly less simple tests", slightlyLessSimpleTests);
-
-  group("more challenging tests", moreChallengingTests);
-
-  /// Edge cases are scenarios that are problems that occur under specific conditions
-  /// Read more: https://en.wikipedia.org/wiki/Edge_case
-  group("edge case tests", edgeCaseTests);
```

Comment:
As one of the early exercises, I believe the purpose here was to inform the learner about Edge cases as they progress in the track/exercise. At some point it even made sense to me to distribute tests cases and group them in the order of their rareness. As this is beneficial in real-world software development. First you want to get the MVP out, then you want to flesh out all the bugs and edge cases.

As this was a hand-written exercise, it had the human element of thinking about the journey of the learner. 

I am curious to know what you think about automating this?

It removes the human element and makes it purely mechanical. Which is useful when you don't have enough human-resources.

Reply:
I think that makes sense, if you'd like to open a PR that suggests the grouping in the `problem-specifications` (or start with an issue, maybe, in case the overwhelming majority don't want to go down that road, then you wouldn't have done the actual work up-front).

Reply:
> My question was more about the fact that by automating, you'd be removing (or at least reducing) the human creativity that each exercise can have.

I 100% agree that we want to introduce people to the language, its idioms, paradigms, etc.

The generator isn't something that is enforced. It's really just a tool, because for most exercises the design work is in deciding what the API to the subject under test is going to look like. The generator is fairly flexible, so we can always tweak what a test suite will look like. Or we can ignore the generator and hand-craft a test suite if it seems like that is better.

In the case of this test suite, there's nothing about the grouping that is inherently specific to Dart (to my mind, anyway). By adding grouping, people can benefit from the pedagogical onramp that you've thought out, or they can ignore it entirely and list out all the tests in a flat structure.

Reply:
@kytrinyx Spot on, if we can't assume someone has prior programming experience, then my intention was to introduce some broader concepts and terminology in the early exercises.

However, with the move to Exercism V3 these practice exercises can be tackled in any order, correct?

And we don't have a curriculum for the non-practice exercises, so assuming my understanding is correct, we'll have to rethink how we bring up these concepts to students.

I apologize for not getting the non-practice exercises created. I thought each track could set up its own path and I would work on the non-practice exercises every so often.

But during that time, changes kept being made to V3 about how things were done and I just got tired of having to refactor and change what I was doing.

So I got a little burned out. And now my time is pretty well spent during the week.

I do hope someone can sit up the non-practice exercises and keep some of these intentions on mine for the students who are coming fresh face into this track.

Reply:
> Spot on, if we can't assume someone has prior programming experience, then my intention was to introduce some broader concepts and terminology in the early exercises.

Got it. That makes a lot of sense.

Overall we do assume that people have prior programming experience if they're using Exercism (even if it's not a whole lot). In the early days of Exercism we really didn't understand who was using Exercism, but now it's clear that Exercism is a tool for learning programming languages, if you already know how to program (at least a little).

> However, with the move to Exercism V3 these practice exercises can be tackled in any order, correct?

Yes, that is true.

> I apologize for not getting the non-practice exercises created.

No apologies necessary! You have no obligation whatsoever to do that; I'm just grateful for all the work that you've put into this track.

> So I got a little burned out.

❤️ I completely understand!

Even if we do set up a syllabus for the Dart track, it will be focused on Dart-specific concepts rather than broader concepts within programming. We're starting to think about how to create courses that are specific to things other than programming languages, but it's a ways out yet.